### ISSUE TITLE
Proposal: 'source_env when .envrc present'

### ISSUE THREAD

**Issue by eval** [direnv/direnv#699]
**Is your feature request related to a problem? Please describe.**
I often use the following pattern to allow for local overrides:
```bash
...bunch of exports
...then at end of .envrc
if [[ -f ".envrc.local" ]]; then
  source_env ".envrc.local"
fi
```

**Describe the solution you'd like**
How about a variant to `source_env` that does exactly that, but is more descriptive:
```bash
source_env_when_present ".envrc.local"
``` 

**Describe alternatives you've considered**
- maybe_source_env
- source_env_if_present
- source_env_(when|if)_found
- `source_env ".envrc.local" <some arg to tell that it should not err>`

**Additional context**
I can provide a PR.

**Comment by zimbatm** [comment]
Good observation. This will be fixed in #714

**Comment by eval** [comment]
@zimbatm 🎉 Thank you!
